lib/thrift/generator.ex

```diff
@@ -93,6 +94,48 @@ defmodule Thrift.Generator do
     end)
   end
 
+  defp resolve_name_collisions(generated_modules) do
+    Enum.reduce(generated_modules, [], fn({name, quoted}, acc) ->
+      Keyword.update(
+        acc,
+        name,
+        quoted,
+        fn(existing) -> resolve_name_collision(name, existing, quoted) end
+      )
+    end)
+  end
+
+  defp resolve_name_collision(name, q1, q2) do
+    {meta1, ast1} = get_meta_and_ast(q1)
+    {meta2, ast2} = get_meta_and_ast(q2)
+
+    context1 = Keyword.get(meta1, :context)
+    context2 = Keyword.get(meta2, :context)
+    combined_ast = [name, [do: {:__block__, [], ast1 ++ ast2}]]
+
+    # the context will be the generating module
+    # we can combine constants into other modules, but no other combinations
+    # furthermore, we want to keep whichever context is _not_ the constant
+    # generator so that subsequent combinations will work properly
+    cond do
```

Comment:
The generator tells us what kind of modules we're asking to combine.  We could theoretically combine any two modules but I could only think of Constant combining with other module types.  All of the other ones would try to do multiple `defstruct`s in the same module.

Except now that I think of it, Enum could probably also merge with other types?